R/prevalence-DELTA_test.R

```diff
@@ -1,701 +1,1136 @@
-#'      sizes** (draw a random split of samples into `n1`/`n2`), recompute (1–4)
-#'      to get \eqn{T_b}.
+#'      \deqn{
+#'        T_{\mathrm{obs}} =
+#'          \frac{\sum_i w_i z_i}{\sqrt{\sum_i w_i^2}} .
+#'      }
+#'
+#'    - If `prev_strat = "decile"`: peptides are binned into deciles of pooled
+#'      prevalence
+#'      \deqn{
+#'        \frac{n_{g1}\hat p_{g1} + n_{g2}\hat p_{g2}}{n_{g1} + n_{g2}},
+#'      }
+#'      a Stouffer statistic \eqn{T_b} is computed within each bin, and the
+#'      **mean** of the bin-level z-scores is reported as
+#'      \eqn{T_{\mathrm{obs}}}.
+#'
+#' 5. **Permutation scheme and p-value.**
+#'    A **two-sided** permutation p-value for the global shift is computed:
+#'
+#'    - **Paired design**: if both groups have measurements for the same
+#'      `subject_id`, each subject’s labels (`g1` ↔ `g2`) are independently
+#'      flipped with probability 1/2 and steps (1–4) are recomputed to obtain
+#'      \eqn{T_b}.
+#'
+#'    - **Unpaired design**: sample labels are shuffled while preserving group
+#'      sizes (random split into `n1` / `n2`), and steps (1–4) are recomputed
+#'      to obtain \eqn{T_b}.
 #'
 #'    With \eqn{B} permutations, the p-value is
-#'    \deqn{ p = \frac{1 + \sum_{b=1}^{B} \mathbf{1}\{ |T_b| \ge |T_{\mathrm{obs}}| \} }{1 + B} , }
-#'    which is the standard *add-one* estimate ensuring non-zero p under the
-#'    global null.
+#'    \deqn{
+#'      p =
+#'      \frac{1 + \sum_{b=1}^{B} \mathbf{1}\{|T_b| \ge |T_{\mathrm{obs}}|\}}
+#'           {1 + B},
+#'    }
+#'    the standard add-one estimator that remains non-zero under the global
+#'    null.
+#'
+#' 6. **Multiplicity.**
+#'    For each `rank`, permutation p-values `p_perm` are adjusted with BH
+#'    (per-rank) to obtain `p_adj_rank`. A simple categorical summary is
+#'    returned in `category_rank_bh`.
+#'
+#' 7. **Optional summaries.**
+#'
+#'    - If `fold_change != "none"` and a column `fold_change` is present in
+#'      `x`, an aggregate of that column over the peptides and subjects in the
+#'      contrast is returned as `fold_change_<mode>` (`mode` = `"sum"`,
+#'      `"mean"`, `"max"`, `"median"`).
+#'
+#'    - If `cross_prev != "none"`, pooled peptide-level prevalences across
+#'      `g1 ∪ g2` are summarized with the same set of reducers (`"sum"`,
+#'      `"mean"`, `"max"`, `"median"`) and returned as `cross_prev_<mode>`.
 #'
-#' 6) **Multiplicity.** For each `rank`, apply BH to `p_perm` → `p_adj_rank`.
+#' **Input requirements.**
 #'
-#' **Notes.**
-#' - If a stratum doesn’t contain exactly two group levels, it’s skipped.
-#' - Progress can be printed every `progress_perm_every` permutations.
+#' - `exist_col` is treated as 0/1 presence.
+#' - There must be **at most one positive** per
+#'   (`subject_id`, `peptide_id`, `group_col`, `group_value`); paired designs
+#'   can have up to two positives across the two group levels. Violations
+#'   trigger an error.
```

Comment:
Can you give super short example here or elaborate more.
I think I get it, but would be great if it's completely obvious what you mean.